compiler/rustc_errors/src/lib.rs

```diff
@@ -316,18 +321,22 @@ impl CodeSuggestion {
                         }
                     }
                     // Add a whole line highlight per line in the snippet.
+                    let len = part.snippet.split('\n').next().unwrap_or(&part.snippet).len();
                     line_highlight.push(SubstitutionHighlight {
-                        start: cur_lo.col.0,
-                        end: cur_lo.col.0
-                            + part.snippet.split('\n').next().unwrap_or(&part.snippet).len(),
+                        start: (cur_lo.col.0 as isize + acc) as usize,
+                        end: (cur_lo.col.0 as isize + acc + len as isize) as usize,
                     });
+                    buf.push_str(&part.snippet);
+                    prev_hi = sm.lookup_char_pos(part.span.hi());
+                    if prev_hi.line == cur_lo.line {
+                        acc += len as isize - (prev_hi.col.0 - cur_lo.col.0) as isize;
+                    }
```

Comment:
(I feel like i wrote this comment before, but I don't see it anywhere. Maybe I forgot to press the button or something. Sorry if it's a duplicate.)

It seems `prev_hi` temporarily represents `cur_hi` here, making this a bit confusing. The `prev_hi.col.0 - cur_lo.col.0` looks like it is calculating the space between spans (in the wrong order), but it's actually calculating the width of the current span. Using `cur_hi.col.0 - cur_lo.col.0` (and only later doing `prev_hi = cur_hi;`) would make that a lot clearer.